test/createAzureResource.test.ts

```diff
@@ -0,0 +1,104 @@
+import { WebSiteManagementClient, WebSiteManagementModels } from 'azure-arm-website';
+import * as crypto from "crypto";
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, DialogResponses, TestAzureAccount, TestUserInput } from 'vscode-azureextensionui';
+import * as constants from '../src/constants';
+import { WebAppProvider } from '../src/explorer/WebAppProvider';
+import { ext } from '../src/extensionVariables';
+import { longRunningTestsEnabled } from './global.test';
+
+suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(1200 * 1000);
+    const resourceGroupsToDelete: string[] = [];
+    const testAccount: TestAzureAccount = new TestAzureAccount();
+    // tslint:disable-next-line:no-backbone-get-set-outside-model
+    const advancedCreationSetting: boolean | undefined = <boolean>vscode.workspace.getConfiguration(constants.extensionPrefix).get('advancedCreation');
+
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(120 * 1000);
+        await testAccount.signIn();
+        ext.tree = new AzureTreeDataProvider(WebAppProvider, 'appService.startTesting', undefined, testAccount);
+        ext.treeView = vscode.window.createTreeView('azureAppService', { treeDataProvider: ext.tree });
+    });
+
+    suiteTeardown(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        await vscode.workspace.getConfiguration(constants.extensionPrefix).update('advancedCreation', advancedCreationSetting, vscode.ConfigurationTarget.Global);
+        this.timeout(1200 * 1000);
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+        ext.tree.dispose();
+    });
+
+    test('Create and Delete New Web App (Basic)', async () => {
+        const appName: string = getRandomHexString().toLowerCase();
+        await vscode.workspace.getConfiguration(constants.extensionPrefix).update('advancedCreation', false, vscode.ConfigurationTarget.Global);
+        const testInputs: string[] = [appName, 'Linux', 'Node.js 10.10 (LTS - Recommended for new apps)'];
```

Comment:
So we have to update this test every time the LTS changes? Can we refactor so that that's not necessary?